samples/nrf9160/lte_ble_gateway/src/main.c

```diff
@@ -57,19 +62,19 @@ enum {
 static atomic_val_t association_requested;
 
 /* Sensor data */
-static struct gps_nmea gps_nmea_data;
-static struct nrf_cloud_sensor_data gps_cloud_data = {
-	.type = NRF_CLOUD_SENSOR_GPS,
-	.tag = 0x1,
-	.data.ptr = gps_nmea_data.buf,
-	.data.len = GPS_NMEA_SENTENCE_MAX_LENGTH,
-};
+struct nrf_modem_gnss_nmea_data_frame gnss_nmea_data;
+static uint32_t gnss_cloud_data_tag = 0x1;
 static atomic_val_t send_data_enable;
 
 /* Structures for work */
 static struct k_work_delayable leds_update_work;
 static struct k_work_delayable connect_work;
 struct k_work_delayable aggregated_work;
+static struct k_work agps_request_work;
+
+enum agps_state { AGPS_STATE_UNSET, AGPS_STATE_REQUESTED, AGPS_STATE_SET };
+
+static enum agps_state agps_state = AGPS_STATE_UNSET;
```

Comment:
Is this really needed? I see that it's used to check whether `nrf_cloud_agps_process()` is called when data is received from the cloud, but just wondering if that's necessary or if the function can just always be called. I think the nRF Cloud team is also working on improving this, so that the nRF Cloud library will be able to detect that the received data is A-GPS data. Ideally the A-GPS data would then be parsed inside the library and injected to the modem, without any actions from the application.

Reply:
No, this is not needed. I'll remove this.